lib/matplotlib/quiver.py

```diff
@@ -372,12 +372,13 @@ def _parse_args(*args):
     X, Y, U, V, C = [None] * 5
     args = list(args)
 
-    # The use of atleast_1d allows for handling scalar arguments while also
-    # keeping masked arrays
-    if len(args) == 3 or len(args) == 5:
+    if len(args) == 2 or len(args) == 4:
+        V = np.atleast_1d(args.pop(-1))
+        U = np.atleast_1d(args.pop(-1))
+    elif len(args) == 3 or len(args) == 5:
```

Comment:
You are increasing the number of lines of code here, and I don't see any benefit to it.  If you left it the way it was, all you would have to do is insert your `_check_array` into each of the lines extracting U, V, and C, like this:
```python
U = np.atleast_1d(cbook._check_array(args.pop(-1)))
```
Better yet, you could modify your checking function to use `np.atleast_1d` instead of `np.asanyarray`, because the former calls the latter internally.  Then in the original you would just replace `np.atleast1d` with `cbook._check_array` in each of the 3 lines.  Much nicer.